x/posts/internal/keeper/keeper_reactions.go

```diff
@@ -0,0 +1,150 @@
+	reactions = append(reactions, reaction)
+	store.Set(key, k.Cdc.MustMarshalBinaryBare(&reactions))
+
+	return nil
+}
+
+// RemovePostReaction removes the reaction from the given user from the post having the
+// given postID. If no reaction with the same value was previously added from the given user, an expError
+// is returned.
+// nolint: interfacer
+func (k Keeper) RemovePostReaction(ctx sdk.Context, postID types.PostID, user sdk.AccAddress, value string) error {
+	store := ctx.KVStore(k.StoreKey)
+	key := types.PostReactionsStoreKey(postID)
+
+	// Get the existing reactions
+	var reactions types.PostReactions
+	k.Cdc.MustUnmarshalBinaryBare(store.Get(key), &reactions)
+
+	// Check if the user exists
+	if !reactions.ContainsReactionFrom(user, value) {
+		return fmt.Errorf("cannot remove the reaction with value %s from user %s as it does not exist",
+			value, user)
+	}
+
+	// Remove and save the reactions list
+	if newLikes, edited := reactions.RemoveReaction(user, value); edited {
+		if len(newLikes) == 0 {
+			store.Delete(key)
+		} else {
+			store.Set(key, k.Cdc.MustMarshalBinaryBare(&newLikes))
+		}
+	}
+
+	return nil
+}
+
+// GetPostReactions returns the list of reactions that has been associated to the post having the given id
+// nolint: interfacer
+func (k Keeper) GetPostReactions(ctx sdk.Context, postID types.PostID) types.PostReactions {
+	store := ctx.KVStore(k.StoreKey)
+
+	var reactions types.PostReactions
+	k.Cdc.MustUnmarshalBinaryBare(store.Get(types.PostReactionsStoreKey(postID)), &reactions)
+	return reactions
+}
+
+// GetReactions allows to returns the list of reactions that have been stored inside the given context
+func (k Keeper) GetReactions(ctx sdk.Context) map[types.PostID]types.PostReactions {
+	store := ctx.KVStore(k.StoreKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.PostReactionsStorePrefix)
+
+	reactionsData := map[types.PostID]types.PostReactions{}
+	for ; iterator.Valid(); iterator.Next() {
+		var postLikes types.PostReactions
+		k.Cdc.MustUnmarshalBinaryBare(iterator.Value(), &postLikes)
+		idBytes := bytes.TrimPrefix(iterator.Key(), types.PostReactionsStorePrefix)
+		postID, err := types.ParsePostID(string(idBytes))
+		if err != nil {
+			// This should never verify
+			panic(err)
+		}
+
+		reactionsData[postID] = postLikes
+	}
+
+	return reactionsData
+}
+
+// -------------
+// --- Reactions
+// -------------
+
+// RegisterReaction allows to register a new reaction for later reference
+func (k Keeper) RegisterReaction(ctx sdk.Context, reaction types.Reaction) {
+	store := ctx.KVStore(k.StoreKey)
+	key := types.ReactionsStoreKey(reaction.ShortCode, reaction.Subspace)
+	store.Set(key, k.Cdc.MustMarshalBinaryBare(&reaction))
+}
+
+// DoesReactionForShortCodeExist checks whether a reaction already exists for the given shortCode, returning it if it does.
+func (k Keeper) DoesReactionForShortCodeExist(ctx sdk.Context, shortcode string, subspace string) (reaction *types.Reaction, exist bool) {
```

Comment:
If we return an additional `bool` value, we should return `types.Reaction` instead of `*types.Reaction` for as the first value. Otherwise, we can omit the `bool` and return a pointer (that will be `null` when the reaction does not exist). 

However, I prefer we return a default `types.Reaction{}` if it does not exist. 

Also, by the name we can simply return the `bool`. Otherwise we might call the method `GetReactionForShortCode` instead.